integration_tests/tests/s3.rs

```diff
@@ -530,6 +530,38 @@ fn test_list_objects_encoding() {
     test_delete_object(&test_client.s3, &bucket_name, &key);
 }
 
+#[test]
+fn test_name_space_truncate() {
+    init_logging();
+
+    let bucket_name = format!("test-name-space-{}", get_time().sec);
+    let test_client = TestS3Client::new(bucket_name.clone());
+
+    test_client.create_test_bucket(bucket_name.clone());
+
+    let filename_spaces = "spaces     ".to_owned();
+    test_client.put_test_object(filename_spaces.clone());
+
+    let req = ListObjectsV2Request {
+        bucket: bucket_name.clone(),
+        ..Default::default()
+    };
+
+    let key = &test_client
+        .s3
+        .list_objects_v2(req)
+        .sync()
+        .unwrap()
+        .contents
+        .unwrap()[0]
+        .clone()
+        .key
+        .unwrap();
+
+    assert_eq!(*key, filename_spaces);
+    test_delete_object(&test_client.s3, &bucket_name, &filename_spaces);
```

Comment:
It seems like this test is [failing in ci](https://semaphoreci.com/matthewkmayer/rusoto/branches/pull-request-1581/builds/3)

```
test test_name_space_truncate ... FAILEDB
test test_puts_gets_deletes_metadata ... okB
test test_puts_gets_deletes ... okB
test test_puts_gets_deletes_utf8 ... okB
test test_multipart_stream_uploads ... okB
test test_puts_gets_deletes_presigned_url ... okB

failures:

---- test_name_space_truncate stdout ----
picked up non-standard endpoint Custom { name: "us-east-1", endpoint: "http://localhost:9000" } from S3_ENDPOINT env. variable
thread 'test_name_space_truncate' panicked at 'assertion failed: `(left == right)`
  left: `"spaces"`,
 right: `"spaces     "`', tests/s3.rs:561:5
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.34/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:339
   9: s3::test_name_space_truncate
  10: s3::test_name_space_truncate::{{closure}}
  11: core::ops::function::FnOnce::call_once
  12: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/625451e376bb2e5283fc4741caa0a3e8a2ca4d54/src/liballoc/boxed.rs:787
  13: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:80
  14: std::panicking::try
             at /rustc/625451e376bb2e5283fc4741caa0a3e8a2ca4d54/src/libstd/panicking.rs:275
  15: std::panic::catch_unwind
             at /rustc/625451e376bb2e5283fc4741caa0a3e8a2ca4d54/src/libstd/panic.rs:394
  16: test::run_test::run_test_inner::{{closure}}
             at src/libtest/lib.rs:1408
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Failed to delete S3 bucket: <?xml version="1.0" encoding="UTF-8"?><Error><Code>BucketNotEmpty</Code><BucketName>test-name-space-1573085690</BucketName><RequestId>tx00000000000000000001f-005dc361fd-101a-default</RequestId><HostId>101a-default-default</HostId></Error>
```

Reply:
Yes saw that. I forget, when updating the service crategen do we also check in the generated files that are affected by that change?

Reply:
So might have to do back to the drawing board if that big of a change wont work.

Reply:
I'll run the generate command on all services and push that change up :crossed_fingers: